db/change_cache.go

```diff
@@ -574,12 +574,41 @@ func (c *changeCache) releaseUnusedSequence(ctx context.Context, sequence uint64
 	} else {
 		changedChannels.Add(unusedSeq)
 	}
-	c.channelCache.AddSkippedSequence(change)
+	c.channelCache.AddUnusedSequence(change)
 	if c.notifyChange != nil && len(changedChannels) > 0 {
 		c.notifyChange(ctx, changedChannels)
 	}
 }
 
+// releaseUnusedSequenceRange calls processEntry for each sequence in the range, but only issues a single notify.
+func (c *changeCache) releaseUnusedSequenceRange(ctx context.Context, fromSequence uint64, toSequence uint64, timeReceived time.Time) {
+
+	var allChangedChannels channels.Set
+	base.InfofCtx(ctx, base.KeyCache, "Received #%d-#%d (unused sequence range)", fromSequence, toSequence)
+	for sequence := fromSequence; sequence <= toSequence; sequence++ {
+		change := &LogEntry{
+			Sequence:     sequence,
+			TimeReceived: timeReceived,
+		}
+
+		// Since processEntry may unblock pending sequences, if there were any changed channels we need
+		// to notify any change listeners that are working changes feeds for these channels
+		changedChannels := c.processEntry(ctx, change)
+		allChangedChannels = allChangedChannels.Update(changedChannels)
+		c.channelCache.AddUnusedSequence(change)
+	}
+
+	unusedSeq := channels.NewID(unusedSeqKey, unusedSeqCollectionID)
+	if allChangedChannels == nil {
+		allChangedChannels = channels.SetOfNoValidate(unusedSeq)
+	} else {
+		allChangedChannels.Add(unusedSeq)
+	}
```

Comment:
Take or leave:

I think this would be easier to read if you initialize this before the loop

```
unusedSeq := channels.NewID(unusedSeqKey, unusedSeqCollectionID)
allChangedChannels := channels.SetOfNoValidate(unusedSeq)
```